### PR TITLE
fix(wfctl): make plugin lock checksums portable

### PR DESCRIPTION
**Pull request by intel352**
## Summary
- stop recording/enforcing host-specific top-level `sha256` in new-format `.wfctl-lock.yaml`
- keep platform `sha256` as archive/download checksum verified by `installFromURL`
- scrub stale top-level checksums before install attempts and fail if scrubbed lockfile persistence fails
- preserve legacy old-format plugin lock behavior outside the new-format path

## TDD / Regression Evidence
With old behavior restored, focused tests failed with binary checksum mismatch on new-format top-level SHA and preserved `existing-sha256` during plugin lock regeneration. With the fix restored, focused tests passed.

## Verification
- `GOWORK=off go test ./cmd/wfctl ./config -run 'Test(InstallFromWfctlLockfile|PluginLock|WfctlLockfile)' -count=1`\n- `GOWORK=off go test ./cmd/wfctl ./config -count=1`\n\n## Review\nAntagonistic reviewer loop reached SHIP-IT. Minor findings were cleaned up before PR: `sha256,omitempty`, precise top-level-key assertion, and clone-before-scrub persistence behavior.

### CODE REVIEW

File: cmd/wfctl/plugin_lock_test.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ plugins:
 	if err != nil {
 		t.Fatalf("read lockfile: %v", err)
 	}
+	if strings.Contains(string(data), "sha256:") {
+		t.Fatalf("regenerated new-format lockfile should not contain top-level sha256:\n%s", data)
+	}
```
[the remark this answers is not in the document]
Comment (intel352) [reply cmd/wfctl/plugin_lock_test.go]: Addressed in 9b11ab0 by replacing the raw sha256 string scan with a top-level plugin map key assertion.

File: cmd/wfctl/plugin_lock_test.go [hunk 2]
```diff
@@ -96,6 +100,15 @@ plugins:
 	if err != nil {
 		t.Fatalf("read lockfile: %v", err)
 	}
+	var raw map[string]any
+	if err := yaml.Unmarshal(data, &raw); err != nil {
+		t.Fatalf("parse raw lockfile: %v", err)
+	}
+	pluginsRaw := raw["plugins"].(map[string]any)
+	fooRaw := pluginsRaw["workflow-plugin-foo"].(map[string]any)
```
[the remark this answers is not in the document]
Comment (intel352) [reply cmd/wfctl/plugin_lock_test.go]: Addressed in 9b11ab0 by adding a helper with ok checks and helpful failures for the raw YAML plugin map lookup.

File: cmd/wfctl/plugin_install_wfctllock.go [hunk 3]
```diff
@@ -28,6 +28,16 @@ func installFromWfctlLockfile(pluginDirVal, lockPath string, lf *config.WfctlLoc
...
 	}
 
+	if lockPath != "" {
+		scrubbed, changed := scrubbedWfctlLockfileTopLevelSHA256(lf)
+		if changed {
+			if err := config.SaveWfctlLockfile(lockPath, scrubbed); err != nil {
+				return fmt.Errorf("persist scrubbed lockfile: %w", err)
+			}
+			*lf = *scrubbed
+		}
```
[the remark this answers is not in the document]
Comment (intel352) [reply cmd/wfctl/plugin_install_wfctllock.go]: Addressed in 9b11ab0 by always persisting a scrubbed new-format lockfile before install and adding a regression test for an explicit empty top-level sha256 key.

File: config/wfctl_lockfile.go [hunk 4]
```diff
@@ -20,9 +20,12 @@ type WfctlLockfile struct {
...
 type WfctlLockPluginEntry struct {
-	Version   string                       `yaml:"version"`
-	Source    string                       `yaml:"source"`
-	SHA256    string                       `yaml:"sha256"`
+	Version string `yaml:"version"`
+	Source  string `yaml:"source"`
+	// SHA256 is deprecated top-level metadata from early new-format lockfiles.
+	// Platform archive checksums live under Platforms; old-format binary
+	// checksums are handled by cmd/wfctl/plugin_lockfile.go.
+	SHA256    string                       `yaml:"sha256,omitempty"`
```
[the remark this answers is not in the document]
Comment (intel352) [reply config/wfctl_lockfile.go:28]: Verified with gofmt in 9b11ab0. The file had no semantic diff after formatting; gofmt keeps the field aligned with the following struct field.
